api/app/controllers/user.controller.ts

```diff
@@ -4,32 +4,43 @@ import { Review, Transcript, User, Wallet } from "../db/models";
 import { calculateWordDiff } from "../utils/review.inference";
 import { Op } from "sequelize";
 import { ReviewAttributes } from "../types/review";
+import { generateJwtToken, verifyGitHubToken } from "../utils/auth";
 
 // Create and Save a new User
-export async function create(req: Request, res: Response) {
+export async function signUp(req: Request, res: Response) {
   // Validate request
-  if (!req?.body?.username) {
-    res.status(400).send({
-      message: "Username can not be empty!",
-    });
-    return;
-  }
+  try {
+    // Validate request
+    if (!req.body.username) {
+      return res.status(400).send({
+        message: "Username can not be empty!",
+      });
+    }
 
-  // Create a User
-  const userDetails = {
-    githubUsername: req.body.username,
-    permissions: req.body.permissions,
-    email: req.body.email,
-  };
+    // Create a User
+    const userDetails = {
+      githubUsername: req.body.username,
+      permissions: req.body.permissions,
+      email: req.body.email,
+      authToken: req.body.githubAuthToken,
```

Comment:
This authToken field has been designated for alby. Remember we plan to change the name to albyAuth token.